Add Hosting Dashboard Domain contact info

paulopmt1 opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of https://linear.app/a8c/issue/DOTDASH-200/add-contact-details-editing-screen

## Proposed Changes

* This adds the Contact Info screen for HD

## Follow-up after we merge this

- Let's remove [shopping-cart implementation of contact form types](https://github.com/Automattic/wp-calypso/blob/trunk/packages/shopping-cart/src/types.ts#L997-L1037) and refactor the code to start using this one
- Let's prepare this component in a way it can be easily reusable on current's `/domains/manage/all/contact-info/edit/{domainName}/{siteSlug}` route
- Implement side-by-side components on upstream Gutenberg DataForm (Slack thread: p1754509095651169-slack-C0982082MSR)
- Let's ensure Calypso reuses everything we copied here, like the `client/dashboard/data/domain-whois.ts` file, so we can avoid having more duplicate code.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* We need to be able to edit contact info in HD

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Acess the Domains contact-info page on HD: http://calypso.localhost:3000/v2/domains/{domainName}/contact-info

<img width="1546" height="1685" alt="image" src="https://github.com/user-attachments/assets/47af0f45-7897-4ab6-b23a-288d49272723" />



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

paulopmt1 committed:
Created the initial structure for the the /contact-info route

paulopmt1 committed:
Added some initial components to the screen

paulopmt1 committed:
Using DataForm for domain Contact details

paulopmt1 committed:
Finished the first interface version

paulopmt1 committed:
Reading Contact information from /whois endpoint

paulopmt1 committed:
Update Contact details data using backend API

paulopmt1 committed:
Added isDurty state for the save button

paulopmt1 committed:
Added basic validation on DataForm

paulopmt1 committed:
Validating data before saving Contact Info

paulopmt1 committed:
Getting supported contries list from our API

grzegorz-cp committed:
My Jetpack: Updating user connection page header and button text. (#105149)

arcangelini committed:
Substack Import: fix flow bug (#105167)

taipeicoder committed:
Only display error message in the expiry column in the Site Overview (#105168)

dzver committed:
Enable summer special (#105169)

arthur791004 committed:
Site Overview: Implement renew now action (#105160)

* Site Overview: Implement renew now action

* Fix types

* Use replaceAll

* Redirect by window.location.href

claudiucelfilip committed:
Increase timeout between atomic transfer and redirect (#105174)

* Increase timeout between atomic transfer and redirect

* Remove unnecessary delay

chriskmnds committed:
Connect Refresh: Extract Gravatar Magic Login into its own space (#104863)

* Connect Refresh: Extract Gravatar Magic Login into its own space

* whole lotta love

* migrate styles

* cleanup. fix publicToken use?

* unfix publicToken use?

* cmon. works now

* fix loginUrl

* lots of cleanup

* fix form outlines

* fix navigation to login

* extract inner components into outer scope

* extract inner components into outer scope

* cleanup. address feedback

* cleanup

* Update client/login/magic-login/gravatar/style.scss

Co-authored-by: Welly Shen <hivoid19@gmail.com>

* address feedback. progress on code-validated

* fix interval not resetting

* fix code validation transition

* Update client/login/magic-login/gravatar/index.tsx

Co-authored-by: Welly Shen <hivoid19@gmail.com>

* Update client/login/magic-login/gravatar/index.tsx

Co-authored-by: Welly Shen <hivoid19@gmail.com>

* Update client/login/magic-login/gravatar/index.tsx

Co-authored-by: Welly Shen <hivoid19@gmail.com>

* cleanup redirect

---------

Co-authored-by: Welly Shen <hivoid19@gmail.com>

chriskmnds committed:
Connect Refresh: Migrate Akismet create-account to unified Signup (#105116)

sbarbosa committed:
Fixes the Promote Post version check hook (#105163)

claudiucelfilip committed:
Allow plugins upload for plans without sftp (#105176)

chriskmnds committed:
Connect Refresh: Migrate VIP create-account to unified Signup (#105132)

claudiucelfilip committed:
Remove Summer Special install plugin feature from Business and Ecommerce (#105177)

paulopmt1 committed:
Updating @wordpress/components and @wordpress/dataviews packages (#105142)

* Updating @wordpress/components and @wordpress/dataviews packages

* Fix plugin icon

* Fix site icon

---------

Co-authored-by: arthur <arthur.chu@automattic.com>

alshakero committed:
Fix font size when rendering ZD HTML (#105180)

fushar committed:
Site settings: add missing learn more links (#105150)

katinthehatsite committed:
Staging Sites Redesign: Update Tooltip and checkbox disable style (#105158)

* Improve styles

* Ensure we have checkbox

* Fix both disabaled and checked

* Fix tooltip position

* Clean comments

* Update message

* Fix typo

* Break tooltip in two lines

* Allow full sync when selective sync is not disabled

---------

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

coder-karen committed:
Hosting Dashboard: Add site logs initial dataviews with data (#105118)

chriskmnds committed:
Connect Refresh: Migrate Jetpack Cloud create-account to unified Signup (#105178)

gcsecsey committed:
add null checks (#105165)

epeicher committed:
Staging Sites: Notify syncing after completing the endpoint call (#105112)

* Staging Sites: Notify syncing after completing the endpoint call

* Add the callback to the pull action too

* Disable sync button immediately after starting syncing

* Check inflight mutations to disable Sync button

* close modal after mutation completes

---------

Co-authored-by: Gergely Csécsey <gcsecsey@gmail.com>

chriskmnds committed:
Connect Refresh: Migrate Studio create-account to unified Signup (#105133)

mirka committed:
Components: Fix ambiguous `rem()` import (#103385)

leonardost committed:
Domains Hosting Dashboard: Create "Add new DNS record" page (#105095)

* (WIP) Create "Add new DNS record" page to Hosting Dashboard

* Create forms to add all supported DNS record types

* Remove some console.logs and refactor some code

* Refactor code splitting the form configuration for each record type in its own file

* Fix field types and add "required" validation

* Add trailing dot automatically for fields that need to be a FQDN

* Use text area component for TXT record data

* Add description field for DNS records

* Remove empty default element from select fields

* Make placeholders translatable

* Reset form data when changing record types

* Navigate to DNS overview page when clicking on "Cancel"

* Remove query that does not belong to this PR

* Refactor DNSRecord type

* Consolidate component in /dns/add/index.tsx

* Rename `AddDNSRecordFormData` to `DNSRecordFormData` and fix form typing

* Remove `isValid` properties from DNS record configs

* Move DNS record configs to types file

* Rename `types.ts` to `dns-record-configs.ts`

* Small refactor

* Replace straight quote mark by a curly quote mark

* Do not translate DNS record types (e.g. A, ALIAS, CNAME)

* Navigate to DNS overview page after adding a record

* Remove translation from CAA tags

* Improve FQDN comment in the data transformation function

* Remove translations from SRV protocol labels

* Translate placeholder strings for DNS records

* Translate page title

* Remove `WPCOMDNSRecord` that was not being used

* Fix type error because ALIAS did not have a `name` property

* Update type names and definitions to align with #105129

* Update type names to keep consistency

* Update mutation function to a more generic one

* Update mutation function to align with #105129

yashwin committed:
A4A > Sign up: Show Onboarding tour in WC flow (#105175)

Initsogar committed:
Jetpack Cloud: skip wpcomJetpackScanAtomicTransfer middleware on Cloud and A4A (#105189)

ivan-ottinger committed:
Staging Sites: Add separate expand button and allow selecting nodes by clicking on them in the Jetpack Backup FileBrowser (#105161)

* Allow selecting files by clicking on them

* Add separate expand button option to file browser component

* Remove unnecessary min-width

* Correct grammar in a comment

* Fix styling

* Add RTL support for chevrons

* Improve checkbox accessibility

* Adjust accessibility

* Remove tab index from the node button only when `showFileCard` option is disabled

* Add `expandDirectoriesOnClick` prop to control directory expansion behavior in file browser

* Introduce separate `handleExpandButtonClick` handler

* Expansion directories only when `expandDirectoriesOnClick` prop is true

* Simplify checkbox change logic

* Improve contents of

---------

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

gcsecsey committed:
Redirect from deletion banner when the staging site is deleted (#105164)

* redirect from deletion banner when the staging site is deleted

* check if query is undefined

* use calypso/state notices instead of snackbar

* clean up comments

arthur791004 committed:
CalloutOverlay: Fix the overlapping issue (#105195)

alshakero committed:
Fix section name in track events (#105196)

m1r0 committed:
Hosting Dashboard: Add domain glue records DataView (#105184)

alshakero committed:
ZD: Fix metadata when creating a ticket (#105183)

sirbrillig committed:
Packages: Add domain-search to tsconfig (#105186)

fushar committed:
Dashboard v2: render expiring purchase in color (#105170)

paulopmt1 committed:
Add support to opt-out 60-day option

paulopmt1 committed:
Add Save contact info warning

wellyshen committed:
Gravatar: Adjust login-related screens for Gravatar-owned services (#105192)

* Gravatar: Adjust login-related screens for Gravatar owned services

* Reuse one more variable

---------

Co-authored-by: Welly Shen <welly.shen@automattic.com>

zaguiini committed:
Domain Dashboard: Render registered date and renewal CTA (#105188)

* Render the header and registered date

* Render the renew now button if eligible

* Wrap domain name if it doesnt fit in the container

* Use purchase properties

* Revert callback changes

paulopmt1 committed:
Add support to State selection from selected country

paulopmt1 committed:
Add support to State selection from selected country

paulopmt1 committed:
Move country and state logic to inside the contact form component

paulopmt1 committed:
Fix re-render issue that removed the focus from text input while typing

p-jackson committed:
Hosting dashboard: Add a custom empty state to the site list (#104544)

Initsogar committed:
Backups Dashboard: add back up now button (#105190)

* Add enqueue and fetch backup functions

* Add siteBackupsQuery

* Add BackupNowButton

* Render BackupNowButton as Backups page header action

* Remove backup up now button icon

* Refactor BackupNowButton: Simplify button content and tooltip handling

- Moved button content and tooltip text into a structured object for better readability and maintainability.
- Updated mutation function to set the enqueue state before calling the backup function.

* Removed unnecessary empty object from the post request

* Refactor backup data layer to entity/collection pattern

gius80 committed:
Domains Hosting Dashboard: DNS records list (#105129)

* Add dns list placeholder

* Add basic data

* Implement data view

* Add value column

* Update section header

* Hide DataView header

* Add actions placeholders

* Add action menu placeholder

* Update dns record list - handle protected records

* Move action to a separate file

* Move fields in a separate file

* Add restore default records actions placeholders

* Add logit to enable/disable dns actions

* Work in progress

* Add delete/edit callback

* Minor fixes after rebase

* Fix edit dns route

* Implement domain restore actions

* Fine tuning

* Fix type issue

* Restore params

* Fix type error

* Restore DnsRecordType

* Fix type

* Address PR review comments

* Fix dropdown

* Address PR review comments

alshakero committed:
Add more tracking information around creating conversations (#105199)

alshakero committed:
Fix double login e2e (#101899)

fushar commented on client/dashboard/app/queries/domain-whois.ts:
```diff
@@ -0,0 +1,34 @@
+		queryFn: () => fetchDomainWhoisValidate( domainName, contactInformation ),
+	} );
+
+export const updateDomainWhoisMutation = ( domainName: string ) =>
```
[nit] if the mutation is literally an "update", we tend not to include it in the mutation at all, to make it symmetric with the query


```suggestion
export const domainWhoisMutation = ( domainName: string ) =>
```

fushar commented on client/dashboard/app/queries/domain-whois.ts:
```diff
@@ -0,0 +1,34 @@
+			transferLock: boolean;
+		} ) => updateDomainWhois( domainName, formData, transferLock ),
+		onSuccess: () => {
+			queryClient.invalidateQueries( { queryKey: [ 'domains', domainName, 'whois' ] } );
```
We can reuse the query

```suggestion
			queryClient.invalidateQueries( domainWhoisQuery( domainName ) );
```

fushar commented on client/dashboard/data/domain.ts:
```diff
@@ -2,10 +2,52 @@ import wpcom from 'calypso/lib/wp';
 	} );
 }
+
+export async function fetchCountryList(): Promise< CountryListItem[] > {
```
I think these don't belong here... what do you think of extracting these to `domain-supported-countries.ts`, both in data and query layers?

fushar commented on client/dashboard/domains/contact-details/index.tsx:
```diff
@@ -0,0 +1,97 @@
+import { useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { get } from 'lodash';
```
Please don't use lodash; we don't want this dependency in the new hosting dashboard 🙈

gius80 replied:
\+ 1 😁

paulopmt1 replied:
Sounds good, I also added a rule for that in the .eslintrc.js file

fushar commented on client/dashboard/domains/contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,342 @@
+	isSubmitting?: boolean;
+}
+
+export default function ContactForm( {
```
Right now, it's not clear what the responsibilities between index and this contact-form files. What if we also define the mutations here, since we're also calling the countries queries here? This is a good existing example I think: https://github.com/Automattic/wp-calypso/blob/6c837d95ffe00c585af872a2b57db9ce0ce0410a/client/dashboard/sites/settings-web-application-firewall/protect-form.tsx

arthur791004 committed:
Site Overview: Implement change site address action (#105191)

* Site Overview: Implement change site address action

* Fix types

* Update eslint

* Address feedback

* Fix display incorrect wpcom domain

fushar commented on client/dashboard/app/routes/domain-routes.ts:
```diff
@@ -142,7 +142,7 @@ export const domainContactInfoRoute = createRoute( {
 	path: 'contact-info',
 } ).lazy( () =>
-	import( '../../sites/domains/placeholder' ).then( ( d ) =>
+	import( '../../domains/contact-details' ).then( ( d ) =>
```
I have the same comment as this: https://github.com/Automattic/wp-calypso/pull/105113#discussion_r2276053800

Basically I think we should rename the directory to `domains/domain-contact-details`, for the sake of consistency.

fushar commented on client/dashboard/data/domain-whois.ts:
```diff
@@ -0,0 +1,94 @@
+
+export function updateDomainWhois(
+	domainName: string,
+	whoisData: any,
```
Instead of `any`, why not use the types we have in `domains/contact-details/types.ts`? Those types then should really belong here in the data layer 🤔

p-jackson committed:
fix(sites): prevent PreviewSizePicker crash from invalid previewSize … (#105206)

* fix(sites): prevent PreviewSizePicker crash from invalid previewSize values

* Move previewSize sanitization to existing `sanitizeView` function

* Simplify code for updating `previewSize` to a valid value

* Use `undefined` instead of `230` in order to explicitly get the default

* Check whether the previewSize is smaller than the smallest value

* Check whether the previewSize is NaN

---------

Co-authored-by: Arun <arun@arun.blog>
Co-authored-by: arthur <arthur.chu@automattic.com>

bogiii committed:
Domains: Create name-servers component (#105062)

* Create name-servers component

katinthehatsite committed:
Marketplace Site Selector: Fix highlight styles within component (#105047)

* Fix styles within component

* Improve syntax

* Remove content-fade

* Cleanup comments

* Remove unnecessary overide

* Keep fade in

* Revert reader changes

* Update colors and fade in

* Improve fade in removal

* Ensure the hover state on badges remain the same

* Only apply the overrides for specific badges

* Fix unintended changes

* Fix unintended change

* Remove redundant styles

---------

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

gius80 commented on client/dashboard/domains/contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,342 @@
+		{
+			id: 'countryCode',
+			label: __( 'Country' ),
+			type: 'select',
```
```suggestion
			type: 'text',
```
I think type refers to the returned value. This should be probably `text` 🤔

hambai committed:
Add the hosting dashboard domain forwarding add/edit form (#105113)

The Hosting Dashboard add and edit form for Domain Forwarding rules. There are a few follow up tasks that we'll fix

gius80 commented on client/dashboard/domains/contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,342 @@
+						}
+						checked={ getValue( { item: data } ) }
+						onChange={ () => onChange( { [ id ]: ! getValue( { item: data } ) } ) }
+					/>
```
```suggestion
					<HStack spacing={ 0 } alignment="start" justify="flex-start">
						<CheckboxControl
							label=""
							checked={ getValue( { item: data } ) }
							onChange={ () => onChange( { [ id ]: ! getValue( { item: data } ) } ) }
						/>
						{ ! hideLabelFromVision && (
							<Text>
								{ createInterpolateElement(
									sprintf(
										/* translators: %s: "what is this?" link */
										__( 'Opt-out of the 60-day transfer lock <link>%s</link>' ),
										__( 'what is this?' )
									),
									{
										link: <InlineSupportLink supportContext="60-day-transfer-lock" />,
									}
								) }
							</Text>
						) }
					</HStack>
```
Since label only support strings, I wonder If we could do something similar, but i'm not sure if this is a good approach. Maybe @fushar has a better idea 😅

paulopmt1 replied:
Good catch, thanks for the suggestion!

allilevine committed:
Newsletter importer: Update billing warning copy (#105203)

* Newsletter importer: Update billing warning copy.

chriskmnds committed:
Connect Refresh: Fix layout issues with Woo DNA/JPC signup (create-account) form (#105156)

bogiii committed:
Domain Search: Update experiment name (#105131)

* Domain Search: Update experiment name

* Enable eligibility only for logged in users

---------

Co-authored-by: Luis Felipe Zaguini <luisfelipezaguini@gmail.com>

paulopmt1 committed:
Add eslint rule to prevent lodash usage

paulopmt1 committed:
Refactor domain-supported contries and states and removed lodash usage

paulopmt1 committed:
Move mutate events to contact-form

paulopmt1 committed:
Getting supported contries list from our API

paulopmt1 committed:
Fix merge conflict

paulopmt1 committed:
Refactor domain-supported contries and states and removed lodash usage

paulopmt1 committed:
Remove unnecessary code after merge

paulopmt1 committed:
Merge branch 'trunk' into add/domain-dashboard-contact-info

paulopmt1 committed:
Renamed contact-details directory to domain-contact-details

paulopmt1 committed:
Update domain data type and removed types file

paulopmt1 committed:
Use contact-form-fields to store them so the main file doesn't get too big

paulopmt1 committed:
Final review changes

fushar commented on client/dashboard/data/domain-whois.ts:
```diff
@@ -0,0 +1,136 @@
+				contactInformation: domainContactDetails,
+				domainNames: [ domainName ],
+			},
+			camelToSnakeCase
```
Interesting... we do this here but not in the `updateDomainWhois()` function above, even though the type is the same (`DomainContactDetails`), is that expected?

paulopmt1 replied:
Good catch, the data wasn't correct. Fixed.

fushar replied:
Sorry, what was fixed? 😄

fushar replied:
Ah, got it. But then `ContactValidationRequestContactInformation`'s fields are already snake case, so why do we need to transform it using `camelToSnakeCase()`?

paulopmt1 replied:
We were using it to convert contactInformation and domainNames, but we can do that by just passing the right properties instead. Good catch, [fixed it here](https://github.com/Automattic/wp-calypso/pull/105075/commits/55b9a17b73b19a1b535c939c4d1f2d0d6e11b973).

fushar commented on client/dashboard/domains/domain-contact-details/contact-form-fields.tsx:
```diff
@@ -0,0 +1,183 @@
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import InlineSupportLink from '../../components/inline-support-link';
+import type { DomainContactDetails } from '../../data/domain-whois';
```
Let's import from `../../data/types` instead, if we're doing it from components.

fushar commented on client/dashboard/data/domain-supported-countries.ts:
```diff
@@ -0,0 +1,44 @@
+	 */
+	tax_name?: string;
+}
+export interface CountryListItemWithoutVat extends CountryListItemBase {
```
[nit] I think we can simplify

- `CountryListItem*` -> `Country*`
- `StatesListItem` -> `State`

paulopmt1 replied:
Good idea, I'll leave it to [when doing this refactor](https://linear.app/a8c/issue/DOTDASH-310/refactor-countrylistitem-type-usage).

fushar commented on client/dashboard/domains/domain-contact-details/contact-form-fields.tsx:
```diff
@@ -0,0 +1,183 @@
+import type { DomainContactDetails } from '../../data/domain-whois';
+
+const createStateFieldEdit = (
+	statesList: Array< { name: string; code: string } > | undefined
```
We can just use the `StatesListItem` type I believe

fushar commented on client/dashboard/domains/domain-contact-details/contact-form.scss:
```diff
@@ -0,0 +1,3 @@
+.contact-form {
```
Remove? 😄

fushar commented on client/dashboard/domains/domain-contact-details/index.tsx:
```diff
@@ -0,0 +1,56 @@
+
+	return (
+		<PageLayout size="small" header={ <PageHeader title={ __( 'Contact details' ) } /> }>
+			<div className="domain-contact-info">
```
Let's use HStack/VStack instead of custom classes. (I believe we can delete it now since it contains no styles)

fushar commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+	);
+	const { data: statesList } = useQuery( statesListQuery( selectedCountryCode ) );
+
+	const formDataRef = useRef< any >( null );
```
I wonder what's this ref for? Seems unused

paulopmt1 replied:
It isn't anymore. Good catch!

fushar commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+	const updateMutation = useMutation( domainWhoisMutation( domainName ) );
+
+	const validateMutation = useMutation( {
+		mutationFn: ( formData: any ) => {
```
We should reuse the existing `domainWhoisValidateQuery()` here. And I believe it should then be a mutation, not query?

p-jackson replied:
Yeah this one is kinda interesting. It is using a `POST`, so mutation seems like the right thing to do. We also want to the notifications to appear `onSuccess`, and that's the sort of thing you use `useMutation()` for. Maybe the function should be called `validateDomainWhois()` rather than `fetchDomainWhoisValidate()`.

But then, why is the `domainWhoisValidateQuery()` using a `POST` request to fetch its data? 😕😢

fushar replied:
> using a POST request to fetch its data? 😕😢

My suspicion is that the payload is so large that using GET might be inconvenient.

> Maybe the function should be called validateDomainWhois() rather than fetchDomainWhoisValidate().

Agree 👍

paulopmt1 replied:
Yeah, definitely not ideal to use `POST` for that. I updated the fucntion to `validateDomainWhois`

fushar commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+						</Notice>
+						<HStack justify="flex-start" spacing={ 2 }>
+							<Button
+								onClick={ () => handleSubmit( formData ) }
```
We should use html `<form>` and put the `handleSubmit` there. See: https://github.com/Automattic/wp-calypso/blob/c5e64fbcc29b5130ddb2398fab45b64d419b62a5/client/dashboard/sites/settings-php/index.tsx#L98

p-jackson commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+		if ( formData.countryCode ) {
+			setSelectedCountryCode( formData.countryCode as string );
+		}
+	}, [ formData.countryCode, setSelectedCountryCode ] );
```
🚨 suspicious looking `useEffect` 🚨

`selectedCountryCode` can be computed and so it doesn't need to be state, and this `useEffect` can be deleted.

The code for fetching the `statesList` could look something like this (after lifting the `formData` state up a bit)

```tsx
const selectedCountryCode = formData.countryCode ?? initialData?.countryCode ?? '';
const { data: statesList } = useQuery( statesListQuery( selectedCountryCode ) );
```

paulopmt1 replied:
Good catch! Thanks for providing an example as well.

p-jackson commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+	const updateMutation = useMutation( domainWhoisMutation( domainName ) );
+
+	const validateMutation = useMutation( {
+		mutationFn: ( formData: any ) => {
```
I don't think this needs to be an `any`. This function accepts a `DomainContactDetails` no?

p-jackson commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+			formDataRef.current = formData;
+			return fetchDomainWhoisValidate( domainName, formData );
+		},
+		onSuccess: ( data: any ) => {
```
We should be able to delete the `any` here. `data` will be whatever type `fetchDomainWhoisValidate()` returns. If accessing `data.success` or `data.messages_simple` is causing errors then it probably means the `fetchDomainWhoisValidate()` types need to be fixed.

p-jackson commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,223 @@
+										{
+											strong: <strong />,
+											agreementlink: (
+												<a
```
We should use `<ExternalLink>` for these links that are opening new tabs

paulopmt1 replied:
I wish we could, do you know a way to not showing the `link__icon` of it? 

I'm looking for this interface:

<img width="702" height="120" alt="image" src="https://github.com/user-attachments/assets/c129cf4d-aad1-4c34-b454-c1a0d86cfb9d" />

If I use the <ExternalLink> [as I did here](https://github.com/Automattic/wp-calypso/blob/add/domain-dashboard-contact-info/client/dashboard/domains/domain-contact-details/contact-form.tsx#L152-L157), it will do this instead:

<img width="698" height="137" alt="image" src="https://github.com/user-attachments/assets/6e8120d6-8fa4-4ab1-86a8-719e7a18cfdf" />

zaguiini replied:
Coming as an outsider, but I think we should use it without modifications. If we have established WPDS components, it doesn't make sense to start customizing it. And I honestly think the version with the arrow is better as it's clear for the user it points elsewhere.

paulopmt1 replied:
After double-checking the original links, we only need one of them to open externally, so it won't be an issue to use the ExternalLink anymore:

<img width="699" height="121" alt="image" src="https://github.com/user-attachments/assets/ca5c4e3f-390e-47d4-8bc9-6f255acf1bd6" />

The Designated Agent will also open an inline support guide.

p-jackson commented on client/dashboard/utils/domain-whois.js:
```diff
@@ -0,0 +1,10 @@
+import { find } from 'lodash';
```
Do we really need to bring in `lodash`? pdaBTz-bn-p2

paulopmt1 replied:
Nops, good catch!

p-jackson commented on client/dashboard/utils/domain-whois.js:
```diff
@@ -0,0 +1,10 @@
+import { find } from 'lodash';
```
I also wonder whether we need a separate utils file just for whois. Could it be in the domains util file?

paulopmt1 replied:
We don't need a separate file. I'll join it.

paulopmt1 committed:
Merge branch 'trunk' into add/domain-dashboard-contact-info

paulopmt1 committed:
Removed unnecessary css files and fixed types

paulopmt1 committed:
Readability improvements

paulopmt1 committed:
Fix data types

paulopmt1 committed:
Improve support links usage

fushar commented on client/dashboard/app/queries/domain-whois.ts:
```diff
@@ -0,0 +1,35 @@
+			formData,
+			transferLock,
+		}: {
+			formData: DomainContactDetails;
```
[SUPER nit]: I think it's better to call this `domainContactDetails` to match with the parameter in the function declaration. Besides "form data" is a concept in the component only.

fushar commented on client/dashboard/domains/domain-contact-details/contact-form.tsx:
```diff
@@ -0,0 +1,207 @@
+	const updateMutation = useMutation( domainWhoisMutation( domainName ) );
+
+	const validateMutation = useMutation( {
+		mutationFn: ( formData: ContactValidationRequestContactInformation ) => {
```
We should put this in the query layer as `domainWhoisValidateMutation()`, replacing the existing now-unused `domainWhoisValidateQuery()`.

paulopmt1 replied:
Fixed it [here](https://github.com/Automattic/wp-calypso/pull/105075/commits/b2dd7d32633f8cd98efeeba6a47c05fce7882e5c) and [here](https://github.com/Automattic/wp-calypso/pull/105075/commits/55b9a17b73b19a1b535c939c4d1f2d0d6e11b973)

paulopmt1 committed:
Improve the usage of queries for validate mutation

paulopmt1 committed:
Fix unnecessary type mapping and remove older query

fushar reviewed (approved):
I think the code is good for now. Thank you!

---

I am having trouble entering the phone number. It seems we only show the simple error 

```
There was an error validating your contact information. The field \"Phone\" is not valid.
```

but I could never manage to enter a valid number... until I check the response which contains the instructions 😄 

<img width="691" height="199" alt="Image" src="https://github.com/user-attachments/assets/eacd5aff-fae0-4188-aa61-7b5c6109fe3b" />

It can be follow-up in a follow-up PR, though.

paulopmt1 commented:
> I am having trouble entering the phone number. It seems we only show the simple error

Good catch! We'll follow up with [a component creation](https://linear.app/a8c/issue/DOTDASH-259/create-a-phone-component) and also the [async validation connection](https://linear.app/a8c/issue/DOTDASH-318/create-async-validation-for-contact-info).

Thanks for the reviews, I'll land this as soon as the tests passes :)

paulopmt1 committed:
Fix type error on StateFieldEdit component

paulopmt1 committed:
Fix render error during component update
